src/packagers/JSPackager.js

```diff
@@ -69,8 +69,20 @@ class JSPackager extends Packager {
           !this.bundle.assets.has(mod) &&
           (!this.bundle.parentBundle || this.bundle.parentBundle.type !== 'js')
         ) {
-          this.externalModules.add(mod);
-          this.bundleLoaders.add(mod.type);
+          if(mod.type === "html") {
+            this.externalModules.add(mod);
+            this.bundleLoaders.add(mod.type);
+
+            // if your JS file imports an html dependency, the html
+            // needs to be included in the generated javascript file
+            await this.writeModule(
+              mod.id,
+              `module.exports=${JSON.stringify(mod.generated.html)}`
```

Comment:
The problem with inlining like this is that we do additional processing on the HTML output in the HTML packager, which wouldn't be accounted for here. For example, we insert additional script/style tags when you import css from js etc. This problem is related to #1456, which is about doing the exact opposite of this and inlining JS/CSS into HTML. We should be able to solve both issues with the same architecture.